projects/akita-ng-fire/src/lib/collection/collection.service.ts

```diff
@@ -253,10 +255,11 @@ export class CollectionService<
     if (syncOptions.loading) {
       setLoading(storeName, true);
     }
+
+    const syncQuery = query(collection(this.db, path), ...queryConstraints);
+
     // Start Listening
-    return this.db
-      .collection<EntityType>(path, queryFn)
-      .stateChanges()
+    return collectionChanges(syncQuery)
```

Comment:
I think the `collectionChanges` takes a generic type. It would be better than casting it later